hudi-spark-datasource/hudi-spark/src/test/java/org/apache/hudi/table/upgrade/TestUpgradeDowngrade.java

```diff
@@ -304,6 +429,27 @@ private static Stream<Arguments> upgradeDowngradeVersionPairs() {
     );
   }
 
+  /**
+   * Test cases for auto-upgrade with different hoodie.write.table.version configurations.
+   * Each case starts with table version SIX and tests different write table version settings.
+   * Test params are: Integer writeTableVersion, HoodieTableVersion expectedVersion, String description
+   */
+  private static Stream<Arguments> writeTableVersionTestCases() {
```

Comment:
nit: put this method before `testAutoUpgradeWithWriteTableVersionConfiguration` so it's easier to check the testing arguments?